model/clusters_mgmt/v1/cluster_autonode_type.model

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright (c) 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The AutoNode configuration for the Cluster.
+struct ClusterAutoNode {
+    // State indicates the current state of AutoNode on this cluster.
+    // Valid values: "enabled", "disabled".
+    State String
+    AWS AwsAutoNodeConfiguration
```

Comment:
This should live under `cluster.AWS`. See https://gitlab.cee.redhat.com/service/api-guidelines/#cloud-specific-components-and-attributes for guidelines on cluster-scoped components that contain cloud-specific configurations.